app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,197 @@
+import androidx.compose.material.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.lib.state.ext.observeAsComposableState
 import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.NimbusMessageCardBinding
-import org.mozilla.fenix.ext.increaseTapArea
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SectionHeader
+import org.mozilla.fenix.compose.button.PrimaryButton
 import org.mozilla.fenix.gleanplumb.Message
 import org.mozilla.fenix.home.sessioncontrol.SessionControlInteractor
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.StyleData
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
 
+/**
+ * View holder for the Nimbus Message Card.
+ *
+ * @property interactor [SessionControlInteractor] which will have delegated to all user
+ * interactions.
+ */
 class MessageCardViewHolder(
-    view: View,
-    private val interactor: SessionControlInteractor
-) : RecyclerView.ViewHolder(view) {
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: SessionControlInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
-    fun bind(message: Message) {
-        val binding = NimbusMessageCardBinding.bind(itemView)
+    companion object {
+        internal val LAYOUT_ID = View.generateViewId()
+    }
+
+    @Composable
+    override fun Content() {
+        val messageGlobal = components.appStore.observeAsComposableState { state -> state.messaging }
+        val messageValue = messageGlobal.value?.messageToShow
 
-        if (message.data.title.isNullOrBlank()) {
-            binding.titleText.isVisible = false
-        } else {
-            binding.titleText.text = message.data.title
+        messageValue?.let {
+            MessageCard(
+                message = it,
+                onMessageClicked = { interactor.onMessageClicked(messageValue) },
+                onMessageClosedClicked = { interactor.onMessageClosedClicked(messageValue) }
+            )
         }
+    }
+}
 
-        binding.descriptionText.text = message.data.text
+/**
+ * Message Card.
+ *
+ * @param message contains title and description of message.
+ * @param onMessageClicked Invoked when user clicks on the button.
+ * @param onMessageClosedClicked Invoked when user clicks on close button to remove message.
+ */
+@Suppress("LongMethod")
+@Composable
+fun MessageCard(
```

Comment:
nit: could we possibly move `MessageCard` and its previews to another file? that way the relocation refactoring is already done by the time we get to killing off all of the viewholder code.